pygef/plot_utils.py

```diff
@@ -137,26 +138,46 @@ def assign_color(df, classification, colors):
     """
     if colors is None:
         if classification == "robertson":
+            df = (
+                df.lazy()
+                .with_column(
+                    pl.col("soil_type")
+                    .apply(lambda row: colours_robertson[row])
+                    .alias("colour")
+                )
+                .collect()
+            )
+
             return (
-                df.assign(
-                    colour=df.apply(
-                        lambda row: colours_robertson[row["soil_type"]], axis=1
-                    )
-                ),
+                df,
                 "Robertson",
             )
         elif classification == "been_jefferies":
+            df = (
+                df.lazy()
+                .with_column(
+                    pl.col("soil_type")
+                    .apply(lambda row: colours_been_jefferies[row])
+                    .alias("colour")
+                )
+                .collect()
+            )
+
             return (
-                df.assign(
-                    colour=df.apply(
-                        lambda row: colours_been_jefferies[row["soil_type"]], axis=1
-                    )
-                ),
+                df,
                 "Been Jefferies",
             )
     else:
+        df = (
+            df.lazy()
```

Comment:
in case of a `lazy().single_op().collect()`, you can just write `df.single_op()`